src/pljson_parser.impl.sql

```diff
@@ -22,25 +22,25 @@ create or replace package body pljson_parser as
   */
 
   decimalpoint varchar2(1 char) := '.';
+  -- Why 222? - I like it.
+  substr_amount number := 222;
```

Comment:
You're actually reducing the performance by increasing the number of read loops since you are reducing the read bytes from 4000 to 222.

Reply:
No.
Main idea
increase productivity pljson_parser.next_char
1. calc length src less often - if (indx > length(s.src) + s.offset_chars) then - it's too often
2. substr symbols to collection - return substr(s.src, indx-s.offset_chars, 1); - it's too often

Reply:
> You're actually reducing the performance by increasing the number of read loops since you are reducing the read bytes from 4000 to 222.

[small_test.txt](https://github.com/pljson/pljson/files/6992799/small_test.txt)
You could try this small test.

Oracle Database 11g Enterprise Edition Release 11.2.0.4.0

PL/JSON 2.4.0
 length - 446 byte; length - 314 char.  time: .23 sec
 length - 3586 byte; length - 2530 char.  time: 3.99 sec
 length - 43054 byte; length - 30382 char.  time: 67.09 sec

PL/JSON 3.5.2
 length - 446 byte; length - 314 char.  time: .32 sec
 length - 3586 byte; length - 2530 char.  time: 8.63 sec
 length - 43054 byte; length - 30382 char.  time: 147.11 sec

PL/JSON pljson_parser_next_char_productivity
 length - 446 byte; length - 314 char.  time: .27 sec
 length - 3586 byte; length - 2530 char.  time: 2.05 sec
 length - 43054 byte; length - 30382 char.  time: 26.96 sec